### PR TITLE
Week 4 assessment completed.

### PR DESCRIPTION
**Pull request by nouxcode**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ ✅ ] JavaScript Coding Challenges
- [ ✅ ] Ruby Coding Challenges
- [ ✅ ] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,29 +22,109 @@ const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "s
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 
+describe("removeShuffle", () => {
```
Comment (elysemontano) [inline code-challenges.test.js:25]: Excellent work on this section Nico!!  Your tests and logic look great!  To level up, try utilizing higher-order functions instead of for loops and consider edge cases for your tests.  Not a requirement, but just a way to stretch yourself even further.

File: interview-questions.md [hunk 2]
```diff
@@ -8,42 +8,42 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: In Ruby a block is a chunk of code. The code within the block is always enclosed in curly braces and is always invoked from a function of the same name as the block and uses the yield statement.
 
-1. How do you extract a value in a Ruby hash?
+5. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: In order to extract a value in a Ruby hash you call the hash followed by a key name using brackets which will extract the value associated with that particular key.
 
-Researched answer:
+Researched answer: In Ruby, a hash is a collection of key-value pairs separated by commas. Calling the hash followed by a key name within brackets grabs the value associated with that key. Each value is assigned to a key using a hash rocket (formerly known as a codice_1).
```
Comment (elysemontano) [inline interview-questions.md:37]: Wonderful work!

File: ruby-challenges.rb [hunk 3]
```diff
@@ -11,6 +11,23 @@
 # Expected output: '221 is odd'
 
 
+def even_or_odd(num)
+  if num.to_i.even?
```
Comment (elysemontano) [inline ruby-challenges.rb:15]: Awesome work in converting to an integer first to cover possible edge cases!

File: ruby-challenges.rb [hunk 4]
```diff
@@ -24,6 +41,17 @@
 
 
 
+def remove_vowel(str)
+  temp_str = str.delete('aeiou')
+  p temp_str.delete('AEIOU')
```
Comment (elysemontano) [inline ruby-challenges.rb:46]: These can be combined into one delete as delete will check and remove all characters included in the parenthesis.

File: ruby-challenges.rb [hunk 5]
```diff
@@ -32,3 +60,18 @@
 # Expected output: 'LEARN is not a palindrome'
 palindrome_tester3 = 'Rotator'
 # Expected output: 'Rotator is a palindrome'
+
+
+def valid_palindrome(str)
+  if str.downcase == str.downcase.reverse
+    p "#{str} is a palindrome."
```
Comment (elysemontano) [inline ruby-challenges.rb:67]: Your logic looks great!  Oftentimes, best practice has the print when invoking the method and not inside the method itself.

**Review by elysemontano — APPROVED** [review]
Really great work on this assessment Nico!  I left a comment in the code challenges to consider if you want to stretch yourself and some comments in Ruby to also consider.  Overall you did a fantastic job!  Keep up the great work 👏
